frontend/src/common.ts

```diff
@@ -743,17 +743,37 @@ export interface Class {
     events: ProgramClassEvent[];
     created_at: Date;
 }
-
-export interface ResidentProgramClassInfo {
-    program_name: string;
+export interface BaseResidentProgramClass {
     class_name: string;
     status: ProgClassStatus;
+    credit_types: string;
+}
+
+export interface ResidentProgramClassInfo extends BaseResidentProgramClass {
+    program_name: string;
     enrollment_status: EnrollmentStatus;
     start_date: string;
     end_date: string;
     attendance_percentage: string;
     class_id: number;
     program_id: number;
+    present_attendance: number;
+    absent_attendance: number;
+}
+
+export interface ResidentProgramClassHistory extends BaseResidentProgramClass {
+    program_name: string;
+    date_status_changed: string;
+}
+
+export interface ResidentProgramClassWeeklySchedule {
+    className: string;
+    startDate: string;
+    endDate: string | null;
+    creditTypes: string;
+    status: string;
+    dateStatusChanged: string;
+    dayOfWeek: number; // 0 = Sunday, 6 = Saturday
 }
```

Comment:
@corypride I don't think this matches the type definition in the backend. But what we need is probably found in `EventCalendar`, especially if we can use `/api/student-calendar`.